src/textual/renderables/background_screen.py

```diff
@@ -0,0 +1,82 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Iterable
+
+from rich.console import Console, ConsoleOptions, RenderableType, RenderResult
+from rich.segment import Segment
+from rich.style import Style
+
+from ..color import Color
+
+if TYPE_CHECKING:
+    from ..screen import Screen
+
+
+class BackgroundScreen:
+    """Applies a color on top of an existing renderable."""
+
+    def __init__(
+        self,
+        screen: Screen,
+        color: Color,
+    ) -> None:
+        """Tints a renderable and removes links / meta.
+
+        Args:
+            renderable: A renderable.
+            color: A color (presumably with alpha).
+        """
+        self.screen = screen
+        self.color = color
+
+    @classmethod
+    def process_segments(
+        cls, segments: Iterable[Segment], color: Color
+    ) -> Iterable[Segment]:
+        """Apply tint to segments and remove meta + styles
+
+        Args:
+            segments: Incoming segments.
+            color: Color of tint.
+
+        Returns:
+            Segments with applied tint.
+
+        """
+        from_rich_color = Color.from_rich_color
+        style_from_color = Style.from_color
+        _Segment = Segment
```

Comment:
Yeah, locals are faster than globals. If you are doing many lookups this can make a difference.

The difference is much less with more recent Pythons. but I think it is still worth it.